roles/lib_openshift/library/oc_csr_client.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+# pylint: disable=missing-docstring
+#
+# Copyright 2018 Red Hat, Inc. and/or its affiliates
+# and other contributors as indicated by the @author tags.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# `oc get --raw /api/v1/nodes/${node}/proxy/healthz`
+
+import base64
+import json
+import OpenSSL
+
+from ansible.module_utils.basic import AnsibleModule
+
+DOCUMENTATION = '''
+---
+module: oc_csr_client
+
+short_description: Retrieve, approve, and verify node client csrs
+
+version_added: "2.4"
+
+description:
+    - Runs various commands to list csrs, approve csrs, and verify nodes are
+      ready.
+
+author:
+    - "Michael Gugino <mgugino@redhat.com>"
+'''
+
+EXAMPLES = '''
+# Pass in a message
+- name: Place credentials in file
+  oc_csr_client:
+    oc_bin: "/usr/bin/oc"
+    oc_conf: "/etc/origin/master/admin.kubeconfig"
+    node_list: "['node1.example.com', 'node2.example.com']"
+'''
+
+CERT_MODE = {'client': 'client auth', 'server': 'server auth'}
+
+
+def run_command(module, command, rc_opts={}):
+    rtnc, stdout, err = module.run_command(command, **rc_opts)
+    if rtnc:
+        result = {'failed': True,
+                  'changed': False,
+                  'msg': str(err),
+                  'state': 'unknown'}
+        module.fail_json(**result)
+    return stdout
+
+
+def get_ready_nodes(module, oc_bin, oc_conf):
+    '''Get list of nodes currently ready vi oc'''
+    # plain output is way less to parse.
+    command = "{} {} get nodes".format(oc_bin, oc_conf)
```

Comment:
An overengineered version would be:
`oc get nodes -o jsonpath="{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}"` to output smth like
`ip-172-18-14-252.ec2.internal:OutOfDisk=False;MemoryPressure=False;DiskPressure=False;PIDPressure=False;Ready=True;`, split it and filter out lines with `Ready=True`.

Reply:
That's incomprehensible and probably just as much splitting and filtering.

Reply:
Going back to past discussions where David had stated that oc output is not guaranteed stable I think we should either get yaml output or a template like Vadim is suggesting. Perhaps not as complicated, but it still needs to be a format that we're sure of so if they change column ordering we're not busted.